magenta-lib/src/main/scala/magenta/graph/package.scala

```diff
@@ -1,15 +1,15 @@
 package magenta
 
 package object graph {
-  implicit class RichNodeSet(nodes: Set[Node]) {
-    def filterDeploymentNodes: Set[DeploymentNode] = nodes.flatMap{
-      case deploymentNode:DeploymentNode => Some(deploymentNode)
+  implicit class RichNodeSet[T](nodes: Set[Node[T]]) {
+    def filterMidNodes: Set[MidNode[T]] = nodes.flatMap{
```

Comment:
Oh man - that's a great function. Thanks for the introduction to the world of `collect`.